analytics/msf4j-analytics/src/main/java/org/wso2/msf4j/analytics/AnalyticUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.msf4j.analytics;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.config.ConfigProviderFactory;
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.config.provider.ConfigProvider;
+import org.wso2.msf4j.analytics.internal.DataHolder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Utils contains utility methods to use analytic purposes
+ */
+public class AnalyticUtils {
+
+    private static final Logger logger = LoggerFactory.getLogger(AnalyticUtils.class);
+    private static final String DEPLOYMENT_YAML_SYS_PROPERTY = "msf4j.conf";
+    private static final String DEPLOYMENT_YAML_FILE = "deployment.yaml";
+
+    /**
+     * Retrieve Configuration Provider Object to read analytic configurations
+     * @return configProvider object
+     */
+    public static ConfigProvider getConfigurationProvider() {
+        ConfigProvider configProvider = DataHolder.getInstance().getConfigProvider();
+        if (configProvider == null) {
+            if (DataHolder.getInstance().getBundleContext() != null) {
+                throw new RuntimeException(
+                        "Failed to populate HTTPMonitoringConfig Configuration. Config Provider is Null.");
+            }
+            //Standalone mode
+            String deploymentYamlPath = System.getProperty(DEPLOYMENT_YAML_SYS_PROPERTY);
+            if (deploymentYamlPath == null || deploymentYamlPath.isEmpty()) {
+                logger.info("System property '" + DEPLOYMENT_YAML_SYS_PROPERTY +
+                        "' is not set. Default deployment.yaml file will be used.");
+                deploymentYamlPath = DEPLOYMENT_YAML_FILE;
+                try (InputStream configInputStream = AnalyticUtils.class.getClassLoader()
+                        .getResourceAsStream(DEPLOYMENT_YAML_FILE)) {
+                    if (configInputStream == null) {
+                        throw new RuntimeException("Couldn't find " + deploymentYamlPath);
+                    }
+                    if (Files.notExists(Paths.get(deploymentYamlPath))) {
```

Comment:
IMO I think we should not copy the deployment YAML configuration but rather indicate the developer that we couldn't find the deployment yaml configuration in the given path (of the system property) and therefore is proceeding with the default deployment yaml configuration. WDYT?

Reply:
yes, we first check the system property, if that is not exists we are proceeding with the default deplyment yaml configuration.
we have to copy the deployment yaml because we cannot access the resource file when we are running the jar file. So we copy default deployment yaml file to certain location and then it refers in the config provider.
In the MSF4JMessageProcessor class, we have done similar thing. please check[1]
1.https://github.com/wso2/msf4j/blob/master/core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java#L95-L124

Reply:
We check the Files.notExists() to avoid copy the configuration each time the application run. configuration should copy only once.